polaris/ocean/model/ocean_model_step.py

```diff
@@ -214,3 +258,148 @@ def _update_ntasks(self):
+
+        return out_options
+
+    def _map_mpaso_to_omega_option(self, option, value):
+        """
+        Map MPAS-Ocean namelist option to Omega equivalent
+        """
+        out_option = option
+        found = False
+
+        assert self.map is not None
+        # traverse the map
+        for entry in self.map:
+            options_dict = entry['options']
+            for mpaso_option, omega_option in options_dict.items():
+                if option == mpaso_option:
+                    found = True
+                    out_option = omega_option
+                    break
+
+            if found:
+                break
+
+        if not found:
+            raise ValueError(f'No mapping found for {option}')
+
+        out_option, out_value = self._map_handle_not(out_option, value)
+
+        return out_option, out_value
+
+    def _map_mpaso_to_omega_configs(self, configs):
+        """
+        Map MPAS-Ocean namelist options to Omega config options
+        """
+        out_configs: Dict[str, Dict[str, str]] = {}
+        not_found = []
+        for section, options in configs.items():
+            for option, mpaso_value in options.items():
+                try:
+                    omega_section, omega_option, omega_value = \
+                        self._map_mpaso_to_omega_section_option(
+                            section=section, option=option, value=mpaso_value)
+                    if omega_section not in out_configs:
+                        out_configs[omega_section] = {}
+                    out_configs[omega_section][omega_option] = omega_value
+                except ValueError:
+                    not_found.append(f'{section}/{option}')
+
+        self._warn_not_found(not_found)
+
+        return out_configs
+
+    def _map_mpaso_to_omega_section_option(self, section, option, value):
+        """
+        Map MPAS-Ocean namelist section and option to Omega equivalent
+        """
+        out_section = section
+        out_option = option
+
+        assert self.map is not None
+
+        section_found = False
+        option_found = False
+
+        # traverse the map
+        for entry in self.map:
+            section_found = False
+            section_dict = entry['section']
+            for mpaso_section, omega_seciton in section_dict.items():
+                if section == mpaso_section:
+                    section_found = True
+                    break
+
+            if not section_found:
+                continue
+
+            options_dict = entry['options']
+            for mpaso_option, omega_option in options_dict.items():
+                if option == mpaso_option:
+                    option_found = True
+                    break
+
+            if option_found:
+                out_section = omega_seciton
+                out_option = omega_option
+                break
+
+        if not option_found:
+            raise ValueError(f'No mapping found for {section}/{option}')
```

Comment:
I don't see any reason not to prefer your suggestion over my code.  I'll make that change.

Reply:
Ah, I see.  The issue is that the map is a list of dictionaries.  We have to iterate through the list (the outer loop) and then loop over options if this list entry has the right MPAS-Ocean section name.  So it's not a simple dictionary in the way you're thinking it might be.

The reason that it's not a simple dictionary is that it's a mapping of both sections and options from MPAS-Ocean to Omega.  I didn't see a good way to do this as a simple dictionary.  There wasn't a nice way to list both the section and option for both the inputs and outputs to the dictionary, especially because of the redundancy.  At the same time, there are some MPAS-Ocean config options that are spread across multiple Omega config sections and visa versa.

So I think we're stuck with this relatively clunky approach unless you have a different data structure for the map itself that you would recommend.

Reply:
See more thorough code changes in other suggestion